java/src/main/java/org/extism/sdk/LibExtism.java

```diff
@@ -15,7 +16,105 @@ public interface LibExtism extends Library {
      */
     LibExtism INSTANCE = Native.load("extism", LibExtism.class);
 
+    interface InternalExtismFunction extends Callback {
+        void invoke(
+                Pointer currentPlugin,
+                ExtismVal.ByReference inputs,
+                int nInputs,
+                ExtismVal.ByReference outputs,
+                int nOutputs,
+                Pointer data
+        );
+    }
+
+    class ExtismVal extends Structure {
+        protected static class ByReference extends ExtismVal implements Structure.ByReference{
+            public ByReference() {
+                super();
+            }
+
+            public ByReference(Pointer ptr) {
+                super(ptr);
+            }
+        }
+
+        public ExtismVal() {
+            super();
+            read();
+        }
+
+        public ExtismVal(Pointer ptr) {
+            super(ptr);
+            read();
+        }
+
+        @Override
+        protected List<String> getFieldOrder() {
+            return Arrays.asList("t", "v");
+        }
+
+        public int t;
+        public ExtismValUnion v;
+    }
+
```

Comment:
Wow this implementation is wild, glad you figured it out 😄

Reply:
I searched and tested but in fact it is no longer useful.